From 18a4dd7285608b834318f4df33401c6763d8104b Mon Sep 17 00:00:00 2001
From: The-going <48602507+The-going@users.noreply.github.com>
Date: Mon, 24 Jan 2022 14:24:40 +0300
Subject: [PATCH 087/153] arm64:dts: FIXME: a64-olinuxino add regulator audio
 mmc

Duplicate nodes appeared as a result of patching and this needs
to be fixed and tested on the board
---
 .../dts/allwinner/sun50i-a64-olinuxino.dts    | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
index 21d0bdc28..636b64a1d 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
@@ -52,6 +52,13 @@ reg_usb1_vbus: usb1-vbus {
 		status = "okay";
 	};
 
+	reg_vcc3v3: vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
 	wifi_pwrseq: wifi_pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		reset-gpios = <&r_pio 0 2 GPIO_ACTIVE_LOW>; /* PL2 */
@@ -99,6 +106,10 @@ &ehci1 {
 	status = "okay";
 };
 
+&mixer0 {
+	status = "okay";
+};
+
 &emac {
 	pinctrl-names = "default";
 	pinctrl-0 = <&rgmii_pins>;
@@ -159,6 +170,16 @@ rtl8723bs: wifi@1 {
 	};
 };
 
+&mmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc2_pins>;
+	vmmc-supply = <&reg_vcc3v3>;
+	bus-width = <8>;
+	non-removable;
+	cap-mmc-hw-reset;
+	status = "okay";
+};
+
 &ohci0 {
 	status = "okay";
 };
@@ -184,6 +205,7 @@ &r_pio {
 	 */
 };
 
+/* FIXME: Duplicate node */
 &pio {
 	vcc-pa-supply = <&reg_dcdc1>;
 	vcc-pb-supply = <&reg_dcdc1>;
@@ -365,6 +387,30 @@ &sound_hdmi {
 	status = "okay";
 };
 
+/* FIXME: Duplicate node
+ * &sound {
+ * 	status = "okay";
+ * 	simple-audio-card,widgets = "Microphone", "Internal Microphone Left",
+ * 				    "Microphone", "Internal Microphone Right",
+ * 				    "Headphone", "Headphone Jack";
+ * 	simple-audio-card,aux-devs = <&codec_analog>;
+ * 	simple-audio-card,routing =
+ * 			"Left DAC", "AIF1 Slot 0 Left",
+ * 			"Right DAC", "AIF1 Slot 0 Right",
+ * 			"INL", "LINEOUT",
+ * 			"INR", "LINEOUT",
+ * 			"Headphone Jack", "HP",
+ * 			"AIF1 Slot 0 Left ADC", "Left ADC",
+ * 			"AIF1 Slot 0 Right ADC", "Right ADC",
+ * 			"Left ADC", "ADC",
+ * 			"Right ADC", "ADC",
+ * 			"Internal Microphone Left", "MBIAS",
+ * 			"MIC1", "Internal Microphone Left",
+ * 			"Internal Microphone Right", "HBIAS",
+ * 			"MIC2", "Internal Microphone Right";
+ * };
+ */
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pb_pins>;
-- 
2.35.3

